From 68d3b1dfbbd2f54bf81986d1d733deaf66658b15 Mon Sep 17 00:00:00 2001
From: Davide Viti <zinosat@tiscali.it>
Date: Thu, 14 Jul 2016 10:14:59 +0100
Subject: [PATCH] rename cmake rootfile

ctest fails on the target, because it cannot find CMake.cmake inside
CMAKE_DATA_DIR (typically set to "/usr/share/cmake-3.0"): all *.cmake
files are removed at build time via the target-finalize rule.

This buildroot-specific patch makes sure ctest looks also for
"Modules/CMake.cmake.ctest" before complaining

[Vincent: tweak patch for 3.6.3]
[James: tweak patch for 3.15.4]

Signed-off-by: Davide Viti <zinosat@tiscali.it>
Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>
Signed-off-by: James Hilliard <james.hilliard1@gmail.com>
---
 Source/cmake.cxx | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Source/cmake.cxx b/Source/cmake.cxx
index 3772f09985..c317e4af38 100644
--- a/Source/cmake.cxx
+++ b/Source/cmake.cxx
@@ -1009,7 +1009,12 @@ int cmake::AddCMakePaths()
     "Path to cpack program executable.", cmStateEnums::INTERNAL);
 #endif
   if (!cmSystemTools::FileExists(
-        (cmSystemTools::GetCMakeRoot() + "/Modules/CMake.cmake"))) {
+        (cmSystemTools::GetCMakeRoot() + "/Modules/CMake.cmake")) &&
+        !cmSystemTools::FileExists(
+          (cmSystemTools::GetCMakeRoot()+"/Modules/CMake.cmake.ctest")
+        )
+     )
+  {
     // couldn't find modules
     cmSystemTools::Error(
       "Could not find CMAKE_ROOT !!!\n"
-- 
2.20.1

